### PR TITLE
fix: Add delete supplement note functionality 

### PR DESCRIPTION
**Pull request by marqueone-ps**
# Description
Adds the ability to remove a note from a case file

Bug # CE-601

## Type of change
- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Manual testing?

---

Thanks for the PR!

Any successful deployments (not always required) will be available below.
[Backend](https://nr-compliance-enforcement-339-backend.apps.silver.devops.gov.bc.ca/) available
[Frontend](https://nr-compliance-enforcement-339-frontend.apps.silver.devops.gov.bc.ca/) available

Once merged, code will be promoted and handed off to following workflow run.
[Main Merge Workflow](https://github.com/bcgov/nr-compliance-enforcement/actions/workflows/merge-main.yml)

### CODE REVIEW

**Review by Scarlett-Truong — CHANGES_REQUESTED** [review]
There's a bug I encountered: when you started create a Note first, without create other Outcome like Assessment or Prevention,  you cannot delete that Note. 

I think when create a note, you can use this `dispatch(setCaseId(res.caseIdentifier));` to update the caseId in the redux store. 

And I just realized some of my codes [here](https://github.com/bcgov/nr-compliance-enforcement/pull/329/files#diff-8f66c21cca11dfdeec23a85d4fb894b0cc0c2a33567f39edd8ae9c79741084c0) are lost in the split-case-reducer refactor :(

**Comment by marqueone-ps** [comment]
@Scarlett-Truong I'm not sure I follow what you mean. When I was testing I was making sure to test when there wasn't any other outcome type so I never ran into a problem deleting. I'm not sure that dispatching the caseId makes sense becasue when I create/update/delete a note one of the things I do is reload the casefile to make sure that the changes get applied. Can you show me what you mean tomorrow?

**Comment by marqueone-ps** [comment]
@Scarlett-Truong I see what you mean now. I've updated the notes to reflect that change.

**Comment by Scarlett-Truong** [comment]
> @Scarlett-Truong I see what you mean now. I've updated the notes to reflect that change.

Have you pushed the update @marqueone-ps ? Cuz I haven't seen any changes 😅

File: frontend/src/app/components/containers/complaints/outcomes/supplemental-notes/supplemental-notes-item.tsx [hunk 1]
```diff
@@ -66,7 +68,14 @@ export const SupplementalNotesItem: FC<props> = ({ notes, action, enableEditMode
         </div>
         <div className="comp-details-right-column">
           <CompTextIconButton
-            id="notes-edit-button"
```
Comment (afwilcox) [inline frontend/src/app/components/containers/complaints/outcomes/supplemental-notes/supplemental-notes-item.tsx:69]: Removing the notes-edit-button id from the Edit button is causing some of the Cypress tests to be skipped as the edit button can't be found.

**Comment by Scarlett-Truong** [comment]
Besides the sonar cloud fail at cypress test, everything looks good to me.

**Review by Scarlett-Truong — APPROVED** [review]

**Review by afwilcox — APPROVED** [review]

**Pull request merged**